src/Components/SignUpPage.jsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from "react";
+import "./signUpPage.scss";
+import { account, ID } from "../lib/appwrite";
+import { validateEmailWithRegex } from "../lib/utils/validateEmailWithRegex";
+import { validatePasswordWithRegex } from "../lib/utils/validatePasswordWithRegex";
+
+export default function SignUpPage() {
+  const [showPassword, setShowPassword] = useState(false);
+  const [nameError, setNameError] = useState(true);
+  const [emailError, setEmailError] = useState(true);
+  const [passwordError, setPasswordError] = useState(true);
+
+  // Toggle Show password Button handler //
+  function togglePassword() {
+    setShowPassword(!showPassword);
+  }
+
+  // Create Form SignUp submit button handler to access the all form data in once. //
+  const signUpButtonHandler = async (event) => {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const formDataEntries = formData.entries();
+    const formDataObject = Object.fromEntries(formDataEntries);
+
+    // Email input validation function call. //
+    const isValidEmail = validateEmailWithRegex(formDataObject.email);
+    setEmailError(isValidEmail);
+
+    //  Password input validation function call. //
+    const isValidPassword = validatePasswordWithRegex(formDataObject.password);
+    setPasswordError(isValidPassword);
+
+    // Name input validation function call. //
+    const nameInputNoWhiteSpace = formDataObject.name.trim().length > 0;
+    setNameError(nameInputNoWhiteSpace);
+
+    if (!isValidEmail || !isValidPassword || !nameInputNoWhiteSpace) {
+      return;
+    }
+
+    // Create AuthUser/ Account in appwrite. By given below method. //
+  };
+
+  return (
+    <div className="signUp">
+      <form className="signUp__form" onSubmit={signUpButtonHandler}>
+        <div className="signUp__wrapper">
+          <h1>Create your account</h1>
+          <div className="signUp__name-input-box">
+            <div className="signUp__name-input-inner-box">
+              <input
+                className="signUp__name-input"
+                type="text"
+                name="name"
+                placeholder="Name"
+                required
+              />
+            </div>
+
+            {!nameError && (
```

Comment:
after changing the above code in the function this should be without ! mark then

Reply:
fixed